packages/@aws-cdk/aws-elasticsearch/lib/domain.ts

```diff
@@ -0,0 +1,1503 @@
+ */
+export interface CognitoOptions {
+  /**
+   * The Amazon Cognito identity pool ID that you want Amazon ES to use for Kibana authentication.
+   */
+  readonly identityPoolId: string;
+
+  /**
+   * A role that allows Amazon ES to configure your user pool and identity pool. It must have the `AmazonESCognitoAccess` policy attached to it.
+   *
+   * @see https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-cognito-auth.html#es-cognito-auth-prereq
+   */
+  readonly role: iam.IRole;
+
+  /**
+   * The Amazon Cognito user pool ID that you want Amazon ES to use for Kibana authentication.
+   */
+  readonly userPoolId: string;
+}
+
+/**
+ * The virtual private cloud (VPC) configuration for the Amazon ES domain. For
+ * more information, see [VPC Support for Amazon Elasticsearch Service
+ * Domains](https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-vpc.html)
+ * in the Amazon Elasticsearch Service Developer Guide.
+ */
+export interface VpcOptions {
+  /**
+   * The list of security groups that are associated with the VPC endpoints
+   * for the domain. If you don't provide a security group ID, Amazon ES uses
+   * the default security group for the VPC. To learn more, see [Security Groups for your VPC]
+   * (https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html) in the Amazon VPC
+   * User Guide.
+   */
+  readonly securityGroups: ec2.ISecurityGroup[];
+
+  /**
+   * Provide one subnet for each Availability Zone that your domain uses. For
+   * example, you must specify three subnet IDs for a three Availability Zone
+   * domain. To learn more, see [VPCs and Subnets]
+   * (https://docs.aws.amazon.com/vpc/latest/userguide/VPC_Subnets.html) in the
+   * Amazon VPC User Guide.
+   */
+  readonly subnets: ec2.ISubnet[];
+}
+
+/**
+ * The minimum TLS version required for traffic to the domain.
+ */
+export enum TLSSecurityPolicy {
+  /** Cipher suite TLS 1.0 */
+  TLS_1_0 = 'Policy-Min-TLS-1-0-2019-07',
+  /** Cipher suite TLS 1.2 */
+  TLS_1_2 = 'Policy-Min-TLS-1-2-2019-07'
+}
+
+/**
+ * Specifies options for fine-grained access control.
+ */
+export interface AdvancedSecurityOptions {
+  /**
+   * ARN for the master user. Only specify this or masterUserName, but not both.
+   *
+   * @default - fine-grained access control is disabled
+   */
+  readonly masterUserArn?: string;
+
+  /**
+   * Username for the master user. Only specify this or masterUserArn, but not both.
+   *
+   * @default - fine-grained access control is disabled
+   */
+  readonly masterUserName?: string;
+
+  /**
+   * Password for the master user.
+   *
+   * You can use `SecretValue.plainText` to specify a password in plain text or
+   * use `secretsmanager.Secret.fromSecretAttributes` to reference a secret in
+   * Secrets Manager.
+   *
+   * @default - A Secrets Manager generated password
+   */
+  readonly masterUserPasswordSecret?: cdk.SecretValue;
```

Comment:
The `secret` suffix gives the impression it has to be an actual `Secret`, which isn't accurate since it can also be a literal string.

```suggestion
  readonly masterUserPassword?: cdk.SecretValue;
```

Reply:
It has to be a CDK SecretValue which shouldn't be a literal string from my understanding of the docs? That said I'm quite happy to drop this.

Reply:
You mean that one shouldn't use `SecretValue.plainText`? You're right, but i don't think that encourages it somehow to do that, we have the types for that. Let's remove it.

Reply:
I have dropped the `Secret` suffix.